frontend/src/lib/Auth.ts

```diff
@@ -18,3 +24,13 @@ export const useAuth = () => {
   const { tokenRef } = useContext(AuthContext);
   return tokenRef.current;
```

Comment:
```suggestion
  const token = jwtDecode<CustomJwtPayload>(tokenRef.current);
  return {
    tokenString: tokenRef.current,
    token: token,
  };
```
I think doing this way would be cleaner but idk how much code would this break
you could close it if you don't want to change this

Reply:
if it really needs to modify to this, i think maybe leave for the last because it required all request header that need `useAuth()`  changed